### ISSUE TITLE
Mintty Themes and Coreutils Dircolors

### ISSUE THREAD

**Issue by meffect** [mintty/mintty#647]
A lot of the themes I tried in https://github.com/iamthad/base16-mintty/tree/master/mintty , my `ls --color=auto` , symbolic links targets are very close to the color of the background. So I can't read them basically. Do I need to also be crafting a ~/.dircolors for each theme? It also makes me wonder if I'm looking at the theme as it was meant to be viewed when using `ls --color=auto'. 

My $LS_COLORS is empty because I force it to be (unless i'm doing it wrong)

```
if [ -f "$HOME/.dircolors" ]; then
    eval "$(dircolors $HOME/.dircolors)"
fi
```
and `$HOME/.dircolors` is an empty file

After I log in , `echo $LS_COLORS` is empty

And still some of the themes I cant see some text color againts the background color

**Comment by mintty** [comment]
If you think those themes are not well-aligned with typical dircolors settings, why don't you submit an issue there?
If you think dircolor settings and themes should be somehow coordinated, maybe you can also address this towards dircolors; I'm not really familiar with it, mintty only provides the platform to adjust themes.
As this is not an intrinsic mintty issue, I'm marking it invalid but leaving it open for discussion.

**Comment by meffect** [comment]
ya I just wanted to somehow open a discussion. I dont think it's an issue with mintty, but I don't know of another way to open discussion other than the `Issues` tab. Is there a better way? 

I opened it here mainly because I was hoping to reach others out there who have used the themes feature of mintty and also ran across this issue, and may have a good solution. 

I need to read up on the themes and colors and linux more. I'll try and post anything beneficial I find out.

I found this website which seems pretty helpful in testing themes https://terminal.sexy/

You can add your `ls --color` to it, and test the output. You can get your specific `ls --color` output by using tmux. You add this `bind H capture-pane -e \; save-buffer ~/tmux.hardcopy \; delete-buffer` to your .tmux.conf. Open `tmux` then do the `ls --color` , then press `control + b` , release, and then `shift H` (capital H). Then you can take the tmux.hardcopy contents and paste them into https://terminal.sexy/ using the `Open Window` `Import Template` menu items.

That's as far as I got with expiritmenting. I was hoping i'd just be able to change mintty themes on the fly, but it seems there's a dependency on $LS_COLORS which must manually be set after a theme is chosen. I was expecting to find `.dircolor` companion  along with many of the github repo's with mintty themes as this is such a major issue, but nothing. So I thought I may be doing something wrong.

I wonder if there is a specific dircolors that I should be using that I can eval which will make all mintty themes appear as the designer intended, or does each theme needs it's on uniquely crafted dircolors

**Comment by mintty** [comment]
>  or does each theme needs it's on uniquely crafted dircolors.

I would say so. You could try to invent a format that includes both, though. For distribution, that is.
Mintty could not automatically apply dircolors settings with a theme as dircolors live in the client process space.

And thanks for the link. I'll mention it along with the Color Scheme Designer, and maybe also offer it in the Options menu. I don't see at first glance, though, how it would help with dircolors; where would I need to enter something?

**Comment by meffect** [comment]
Ya good point. It doesn't really help with dircolors, since the ls --color output seems hardcoded in the tmux buffer rather than using color indexes. This is about the point I left off this weekend and need to learn more about it

**Comment by mintty** [comment]
Actually, I changed my mind about regarding terminal.sexy as an alternative theme designer.
While its interface is nicely done, it's basically only a colour value adjuster per ANSI colour.
The Color Scheme Designer (now linked from mintty Options) in contrast offers global configuration parameters that change all colours at once, consistently. This is the special and charming feature of it.

**Comment by meffect** [comment]
does mintty not use ANSI color? whats the alternative to ANSI color?  I'm fairly confused on the ANSI Color, 4 Bit Color, 8 Bit Color, the 16 color indexes, etc differences

**Comment by meffect** [comment]
This is the issue I have not being able to read certain text. it happens with a lot of themes i've tried

![theme](https://cloud.githubusercontent.com/assets/1799758/24374716/bd1b13f6-12ea-11e7-81d9-3436389c3829.png)

![symtargets](https://cloud.githubusercontent.com/assets/1799758/24374755/d9d79f78-12ea-11e7-8a55-2c86d0f8ed46.png)

```
$ cat base16-solarized-dark.minttyrc
ForegroundColour=147,161,161
BackgroundColour=0,43,54
CursorColour=147,161,161
Black=0,43,54
BoldBlack=101,123,131
Red=220,50,47
BoldRed=203,75,22
Green=133,153,0
BoldGreen=7,54,66
Yellow=181,137,0
BoldYellow=88,110,117
Blue=38,139,210
BoldBlue=131,148,150
Magenta=108,113,196
BoldMagenta=238,232,213
Cyan=42,161,152
BoldCyan=211,54,130
White=147,161,161
BoldWhite=253,246,227
```

**Comment by mintty** [comment]
> does mintty not use ANSI color?

Sure it does. What I meant to address is this: To define a "colour scheme", you would define 8 or 16 ANSI colours. Often you define them all separately. The "Color Scheme Designer" offers a tool to modify them unisono, e.g. by changing all 16 brightness values at once, correspondingly. It's not the resulting scheme format that distinguishes that tool, but the method of tuning it.

**Comment by meffect** [comment]
Ohhh ok, got it

**Comment by klauern** [comment]
Okay, so I'm having this same problem (using `wsltty`), and I don't understand whether there's something I'm supposed to do in order to get one of these working.  @meffect, did you ever figure out what the problem was and how to fix it?

![image](https://cloud.githubusercontent.com/assets/4735/24569680/ddc76fe2-162c-11e7-8ccb-c1bec77b882e.png)

**Comment by Cheezmeister** [comment]
Using Debian's `colortest`:

![image](https://cloud.githubusercontent.com/assets/576853/25546265/20fce724-2c17-11e7-97af-1dad7daf88eb.png)

We see that the bolded values for green, yellow, magenta, and blue are actually shades of grey. In this example it's the [Monokai](https://github.com/iamthad/base16-mintty/blob/master/mintty/base16-monokai.minttyrc) theme, but many (most?) of the base16 themes share the same issue.

I'm pretty sure the bug @meffect opened in the base16 repo is valid. For now, I'm going to monkey-patch the theme and move on with my life. For others using WSLtty you'll find it in `<WSLTTY>usr\share\mintty\themes`

**Comment by mintty** [comment]
You are right, that theme is buggy. I should remove the link to that repository from the Tips Wiki page, but I'll allow some time for fixing...

**Comment by mintty** [comment]
Released 2.7.7 with removed entry from Wiki.
